### PR TITLE
Frontend refactor transaction

### PR DESCRIPTION
**Pull request by thisconnect**
builds on https://github.com/digitalbitbox/bitbox-wallet-app/pull/2546

### CODE REVIEW

**Comment by benma** [comment]
Maybe https://github.com/digitalbitbox/bitbox-wallet-app/pull/2547 is still useful, you could re-use the Note component from there to make the Transaction component smaller. wdyt?

**Comment by thisconnect** [comment]
rebased

**Comment by thisconnect** [comment]
rebased

**Comment by thisconnect** [comment]
> And one more thing: WDYT while we're here, we separate transaction.tsx into different smaller components?

makes sense, i.e. I wanted to refcator dialog into its own component. but I didn't want to make this PR bigger and planned to have a new PR for this.

File: frontends/web/src/components/transactions/transaction.tsx [hunk 1]
```diff
@@ -30,176 +30,155 @@ import { Note } from './note';
...
+  ) : type === 'receive' ? (
+    <ArrowIn />
+  ) : type === 'send' ? (
+    <ArrowOut />
+  ) : (
+    <ArrowSelf />
+  );
+  const sign = ((type === 'send') && '−') || ((type === 'receive') && '+') || '';
+  const typeClassName = (status === 'failed' && style.failed) || (type === 'send' && style.send) || (type === 'receive' && style.receive) || '';
+  const sDate = time ? parseTimeShort(time) : '---';
```
[the remark this answers is not in the document]
Comment (thisconnect) [reply frontends/web/src/components/transactions/transaction.tsx:90]: No idea, I guess "short date"?

seems not relevant to this PR class->functional refactoring. I will take a look in a different PR ok?

File: frontends/web/src/components/transactions/transaction.tsx [hunk 2]
```diff
@@ -30,176 +30,155 @@ import { Note } from './note';
...
-                </span>
-              </div>
-            )}
-            <div className={[parentStyle.action, parentStyle.hideOnMedium].join(' ')}>
-              <button type="button" className={style.action} onClick={this.showDetails}>
-                <ExpandIcon expand={!transactionDialog} />
-              </button>
-            </div>
+  return (
+    <div className={[style.container, index === 0 ? style.first : ''].join(' ')}>
```
[the remark this answers is not in the document]
Comment (thisconnect) [reply frontends/web/src/components/transactions/transaction.tsx:95]: yes 💯  but I think for now we  should just merge as it was and I can open a new PR soon™

**Pull request merged**